website_user_login_redirect_custom/controllers/main.py

```
@@ -0,0 +1,62 @@
+
+# Import the base web controller (always available)
+from odoo.addons.web.controllers.main import Home
+
+
+class WebsiteRedirectCustom(Home):
+    """
+    Inherit from Home to override login behavior
+    This works even without auth_signup module
+    """
+
+    def _should_redirect_custom(self):
```

Ruff (0.14.0)

Missing return type annotation for private function `_should_redirect_custom`

(ANN202)